components/feature/qr/src/main/java/mozilla/components/feature/qr/CustomOverlay.kt

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.qr
+
+import android.app.Activity
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.PorterDuff
+import android.graphics.PorterDuffXfermode
+import android.graphics.RectF
+import android.util.AttributeSet
+import android.util.DisplayMetrics
+import android.widget.ImageView
+import android.widget.LinearLayout
+
+/**
+ * A custom overlay [View] that sets the opacity of the background to dim it.
+ * This gives a focus like look to the qr reticle.
+ * @property windowFrame a reference to a [Bitmap], used to create the window frame.
+ * @property activity a reference to an [Activity], used to refer the child views of
+ * calling parent view.
+ * @property ALPHA defines the opacity of the background.
+ * @property RADIUS defines the radius of the rectangle corners.
+ */
+
+class CustomOverlay : LinearLayout {
+    private var windowFrame: Bitmap? = null
+    private lateinit var activity: Activity
+
+    companion object {
+        const val RADIUS = 30.0f
+        const val ALPHA = 170
+    }
+
+    constructor(context: Context) : super(context)
+
+    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
+        activity = context as Activity
+    }
+
+    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
+
+    /**
+     * This is called before the canvas is drawn. Initializes the windowFrame only once
+     * @param canvas The canvas to draw upon.
+     */
+    override fun dispatchDraw(canvas: Canvas) {
+        super.dispatchDraw(canvas)
+
+        if (windowFrame == null) {
+            createWindowFrame()
+        }
+        canvas.drawBitmap(windowFrame!!, 0f, 0f, null)
```

Comment:
nit: You can simplify the null check by having `createWindowFrame` return the newly created windowFrame.

```suggestion
        canvas.drawBitmap(windowFrame ?: createWindowFrame(), 0f, 0f, null)
```

Reply:
I think you meant 
```java
windowFrame ?: createWindowFrame()
canvas.drawBitmap(windowFrame!!, 0f, 0f, null)
```
Because ``createWindowFrame()`` is a void fun